NewRelic exporter fails when no service name is provided

**Describe the bug**
User [report](https://gitter.im/open-telemetry/opentelemetry-service?at=5f549815d4f0f55ebb05e480) on gitter indicated that NewRelic exporter fails. The quick investigation suggests this might happen when [Service Name is not provided](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/master/exporter/newrelicexporter/newrelic.go#L88). 

**Steps to reproduce**
Send a span with no service name

**What did you expect to see?**
No segfault

**What did you see instead?**
Segfault

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x28 pc=0x2c86ba3]

goroutine 94 [running]:
github.com/open-telemetry/opentelemetry-collector-contrib/exporter/newrelicexporter.exporter.pushTraceData(0xc000498780, 0xc0007e0000, 0x3e8bc00, 0xc0007e7340, 0xc00033fe20, 0x0, 0xc000529688, 0xc42f12)
C:/OpenTelemetryWork/opentelemetry-collector-contrib/exporter/newrelicexporter/newrelic.go:88 +0x423
go.opentelemetry.io/collector/exporter/exporterhelper.(*tracesRequest).export(0xc0004b28e0, 0x3e8bc00, 0xc0007e7340, 0x3e8bc00, 0xc0007e7340, 0xc00055a100)
```

**What version did you use?**
Windows, apparently v.0.9.0, though [simple change in unit test](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/master/exporter/newrelicexporter/newrelic_test.go#L63) reproduces this on any platform at master.

**What config did you use?**
```
receivers:
  otlp/withendpoint:
    protocols:
      grpc:
        endpoint: 127.0.0.1:55680
exporters:
  logging:
    logLevel: debug
  newrelic:
    apikey: my-API-key
  zipkin:
    endpoint: "http://localhost:9411/api/v2/spans"
service:
  pipelines:
    traces:
      receivers: [otlp/withendpoint]
      exporters: [logging, newrelic, zipkin]
```

**Environment**
OS: any OS

**Additional context**
It is not entirely clear if a default service name (as it happens for e.g. Zipkin exporter) or no service name needs to be exported in such scenario - maintainer input needed

--- a/exporter/newrelicexporter/newrelic.go
+++ b/exporter/newrelicexporter/newrelic.go
@@ -84,8 +84,14 @@ func (e exporter) pushTraceData(ctx context.Context, td pdata.Traces) (int, erro
 
 	octds := internaldata.TraceDataToOC(td)
 	for _, octd := range octds {
+
+		var srv string
+		if octd.Node != nil && octd.Node.ServiceInfo != nil {
+			srv = octd.Node.ServiceInfo.Name
+		}
+
 		transform := &transformer{
-			ServiceName: octd.Node.ServiceInfo.Name,
+			ServiceName: srv,
 			Resource:    octd.Resource,
 		}
 
@@ -115,9 +121,14 @@ func (e exporter) pushMetricData(ctx context.Context, md pdata.Metrics) (int, er
 
 	ocmds := internaldata.MetricsToOC(md)
 	for _, ocmd := range ocmds {
+		var srv string
+		if ocmd.Node != nil && ocmd.Node.ServiceInfo != nil {
+			srv = ocmd.Node.ServiceInfo.Name
+		}
+
 		transform := &transformer{
 			DeltaCalculator: e.deltaCalculator,
-			ServiceName:     ocmd.Node.ServiceInfo.Name,
+			ServiceName:     srv,
 			Resource:        ocmd.Resource,
 		}
 

--- a/exporter/newrelicexporter/newrelic_test.go
+++ b/exporter/newrelicexporter/newrelic_test.go
@@ -50,14 +50,52 @@ func TestLogWriter(t *testing.T) {
 	assert.Len(t, messages, 2)
 }
 
-func TestExportTraceData(t *testing.T) {
+func testTraceData(t *testing.T, expected []Span, td consumerdata.TraceData) {
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 
-	m := &Mock{make([]Data, 0, 3)}
+	m := &Mock{make([]Data, 0, 1)}
 	srv := m.Server()
 	defer srv.Close()
 
+	f := NewFactory()
+	c := f.CreateDefaultConfig().(*Config)
+	c.APIKey, c.SpansURLOverride = "1", srv.URL
+	params := component.ExporterCreateParams{Logger: zap.NewNop()}
+	exp, err := f.CreateTraceExporter(context.Background(), params, c)
+	require.NoError(t, err)
+	require.NoError(t, exp.ConsumeTraces(ctx, internaldata.OCToTraceData(td)))
+	require.NoError(t, exp.Shutdown(ctx))
+	assert.Equal(t, expected, m.Spans())
+}
+
+func TestExportTraceDataMinimum(t *testing.T) {
+	td := consumerdata.TraceData{
+		Spans: []*tracepb.Span{
+			{
+				TraceId: []byte{1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1},
+				SpanId:  []byte{0, 0, 0, 0, 0, 0, 0, 1},
+				Name:    &tracepb.TruncatableString{Value: "root"},
+			},
+		},
+	}
+
+	expected := []Span{
+		{
+			ID:      "0000000000000001",
+			TraceID: "01010101010101010101010101010101",
+			Attributes: map[string]interface{}{
+				"collector.name":    name,
+				"collector.version": version,
+				"name":              "root",
+			},
+		},
+	}
+
+	testTraceData(t, expected, td)
+}
+
+func TestExportTraceDataFullTrace(t *testing.T) {
 	td := consumerdata.TraceData{
 		Node: &commonpb.Node{
 			ServiceInfo: &commonpb.ServiceInfo{Name: "test-service"},
@@ -90,15 +128,6 @@ func TestExportTraceData(t *testing.T) {
 			},
 		},
 	}
-
-	f := NewFactory()
-	c := f.CreateDefaultConfig().(*Config)
-	c.APIKey, c.SpansURLOverride = "1", srv.URL
-	params := component.ExporterCreateParams{Logger: zap.NewNop()}
-	exp, err := f.CreateTraceExporter(context.Background(), params, c)
-	require.NoError(t, err)
-	require.NoError(t, exp.ConsumeTraces(ctx, internaldata.OCToTraceData(td)))
-	require.NoError(t, exp.Shutdown(ctx))
 
 	expected := []Span{
 		{
@@ -138,10 +167,10 @@ func TestExportTraceData(t *testing.T) {
 		},
 	}
 
-	assert.Equal(t, expected, m.Spans())
+	testTraceData(t, expected, td)
 }
 
-func TestExportMetricData(t *testing.T) {
+func testExportMetricData(t *testing.T, expected []Metric, md consumerdata.MetricsData) {
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 
@@ -149,6 +178,76 @@ func TestExportMetricData(t *testing.T) {
 	srv := m.Server()
 	defer srv.Close()
 
+	f := NewFactory()
+	c := f.CreateDefaultConfig().(*Config)
+	c.APIKey, c.MetricsURLOverride = "1", srv.URL
+	params := component.ExporterCreateParams{Logger: zap.NewNop()}
+	exp, err := f.CreateMetricsExporter(context.Background(), params, c)
+	require.NoError(t, err)
+	require.NoError(t, exp.ConsumeMetrics(ctx, internaldata.OCToMetrics(md)))
+	require.NoError(t, exp.Shutdown(ctx))
+	assert.Equal(t, expected, m.Metrics())
+}
+
+func TestExportMetricDataMinimal(t *testing.T) {
+	desc := "physical property of matter that quantitatively expresses hot and cold"
+	unit := "K"
+	md := consumerdata.MetricsData{
+		Metrics: []*metricspb.Metric{
+			{
+				MetricDescriptor: &metricspb.MetricDescriptor{
+					Name:        "temperature",
+					Description: desc,
+					Unit:        unit,
+					Type:        metricspb.MetricDescriptor_GAUGE_DOUBLE,
+					LabelKeys: []*metricspb.LabelKey{
+						{Key: "location"},
+						{Key: "elevation"},
+					},
+				},
+				Timeseries: []*metricspb.TimeSeries{
+					{
+						LabelValues: []*metricspb.LabelValue{
+							{Value: "Portland", HasValue: true},
+							{Value: "0", HasValue: true},
+						},
+						Points: []*metricspb.Point{
+							{
+								Timestamp: &timestamppb.Timestamp{
+									Seconds: 100,
+								},
+								Value: &metricspb.Point_DoubleValue{
+									DoubleValue: 293.15,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	expected := []Metric{
+		{
+			Name:      "temperature",
+			Type:      "gauge",
+			Value:     293.15,
+			Timestamp: int64(100 * time.Microsecond),
+			Attributes: map[string]interface{}{
+				"collector.name":    name,
+				"collector.version": version,
+				"description":       desc,
+				"unit":              unit,
+				"location":          "Portland",
+				"elevation":         "0",
+			},
+		},
+	}
+
+	testExportMetricData(t, expected, md)
+}
+
+func TestExportMetricDataFull(t *testing.T) {
 	desc := "physical property of matter that quantitatively expresses hot and cold"
 	unit := "K"
 	md := consumerdata.MetricsData{
@@ -234,15 +333,6 @@ func TestExportMetricData(t *testing.T) {
 		},
 	}
 
-	f := NewFactory()
-	c := f.CreateDefaultConfig().(*Config)
-	c.APIKey, c.MetricsURLOverride = "1", srv.URL
-	params := component.ExporterCreateParams{Logger: zap.NewNop()}
-	exp, err := f.CreateMetricsExporter(context.Background(), params, c)
-	require.NoError(t, err)
-	require.NoError(t, exp.ConsumeMetrics(ctx, internaldata.OCToMetrics(md)))
-	require.NoError(t, exp.Shutdown(ctx))
-
 	expected := []Metric{
 		{
 			Name:      "temperature",
@@ -326,5 +416,5 @@ func TestExportMetricData(t *testing.T) {
 		},
 	}
 
-	assert.Equal(t, expected, m.Metrics())
+	testExportMetricData(t, expected, md)
 }

--- a/exporter/newrelicexporter/transformer.go
+++ b/exporter/newrelicexporter/transformer.go
@@ -228,7 +228,9 @@ func (t *transformer) MetricAttributes(metric *metricspb.Metric) map[string]inte
 
 	attrs[collectorNameKey] = name
 	attrs[collectorVersionKey] = version
-	attrs[serviceNameKey] = t.ServiceName
+	if t.ServiceName != "" {
+		attrs[serviceNameKey] = t.ServiceName
+	}
 
 	return attrs
 }

